src/Authentication/BasicApiKey/Server.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * API Key authentication server.
+ *
+ * @package SatisPress
+ * @license GPL-2.0-or-later
+ * @since 0.3.0
+ */
+
+declare ( strict_types = 1 );
+
+namespace SatisPress\Authentication\BasicApiKey;
+
+use SatisPress\Authentication\AbstractServer;
+use SatisPress\Authentication\ApiKey;
+use WP_Error;
+use WP_Http as HTTP;
+use WP_User;
+
+/**
+ * API Key authentication server class.
+ *
+ * @since 0.3.0
+ */
+class Server extends AbstractServer {
+	/**
+	 * Handle authentication.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @param integer|bool $user_id Current user ID or false if unknown.
+	 * @return integer|bool A user on success, or false on failure.
+	 */
+	public function authenticate( $user_id ) {
+		if ( ! empty( $user_id ) || ! $this->should_attempt ) {
+			return $user_id;
+		}
+
+		$header = $this->request->get_header( 'authorization' );
+
+		// Bail if the authorization header doesn't exist.
+		if ( empty( $header ) || 0 !== stripos( $header, 'basic ' ) ) {
+			return $user_id;
+		}
+
+		// The password field isn't used for API Key authentication.
+		$realm = $this->request->get_header( 'PHP_AUTH_PW' );
+
+		// Bail if this isn't a SatisPress authentication request.
+		if ( 'satispress' !== $realm ) {
+			return $user_id;
+		}
+
+		$this->should_attempt = false;
+
+		$api_key_id = $this->request->get_header( 'PHP_AUTH_USER' );
+
+		// Bail if an API Key wasn't provided.
+		if ( empty( $api_key_id ) ) {
+			$this->auth_status = new WP_Error(
```

Comment:
The repetitive creation of these WP_Error objects seems clumsy. Could they either be treated as Exceptions, or otherwise create a `SatisPress\WP_Error` class which extends `WP_Error` with named constructors, like we do with Exceptions?

Then it would be something like:

```php
if ( empty( $api_key_id ) ) {
    $this->auth_status = SatisPress\WP_Error::missingAuthorizationHeader();

    return false;
}

Reply:
It gets even worse for more complex authentication methods! But yeah, I like that approach for the exceptions, so think it could work well here, too.

I've also been thinking about renaming the `ExceptionInterface` to `SatisPressException` to make it more clear what's being caught in a few places. Got any thoughts on that?

Reply:
> I've also been thinking about renaming the ExceptionInterface to SatisPressException to make it more clear what's being caught in a few places. Got any thoughts on that?

Sounds fine with me.